gammapy/data/data_store.py

```diff
@@ -335,3 +392,216 @@ def check_observations(self):
+        info["DATE-OBS"] = header["DATE_OBS"]
+        info["TIME-OBS"] = header["TIME_OBS"]
+        info["DATE-END"] = header["DATE_END"]
+        info["TIME-END"] = header["TIME_END"]
+        info["N_TELS"] = header["N_TELS"]
+        info["OBJECT"] = header["OBJECT"]
+
+        # This is the info needed to link from EVENTS to IRFs
+        info["TELESCOP"] = header["TELESCOP"]
+        info["CALDB"] = header["CALDB"]
+        info["IRF"] = header["IRF"]
+
+        # Not part of the spec, but good to know from which file the info comes
+        info["EVENTS_FILENAME"] = filename
+        info["EVENT_COUNT"] = header["NAXIS2"]
+
+        # gti = Table.read(filename, hdu='GTI')
+        # info['GTI_START'] = gti['START'][0]
+        # info['GTI_STOP'] = gti['STOP'][0]
+
+        return info
+
+    def make_obs_table(self):
+        rows = []
+        for path in self.paths:
+            row = self.get_obs_info(path)
+            rows.append(row)
+
+        names = list(rows[0].keys())
+        table = Table(rows=rows, names=names)
+
+        table["RA_PNT"].unit = "deg"
+        table["DEC_PNT"].unit = "deg"
+        table["GLON_PNT"].unit = "deg"
+        table["GLAT_PNT"].unit = "deg"
+        table["ZEN_PNT"].unit = "deg"
+        table["ALT_PNT"].unit = "deg"
+        table["AZ_PNT"].unit = "deg"
+        table["ONTIME"].unit = "s"
+        table["LIVETIME"].unit = "s"
+        table["TSTART"].unit = "s"
+        table["TSTOP"].unit = "s"
+
+        meta = table.meta
+
+        # TODO: Values copied from one of the EVENTS headers
+        # TODO: check consistency for all EVENTS files and handle inconsistent case
+        # Transform times to first ref time? Or raise error for now?
+        # Test by combining some HESS & CTA runs?
+        meta["MJDREFI"] = 51544
+        meta["MJDREFF"] = 5.0000000000e-01
+        meta["TIMEUNIT"] = "s"
+        meta["TIMESYS"] = "TT"
+        meta["TIMEREF"] = "LOCAL"
+
+        meta["HDUCLASS"] = "GADF"
+        meta[
+            "HDUDOC"
+        ] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+        meta["HDUVERS"] = "0.2"
+        meta["HDUCLAS1"] = "INDEX"
+        meta["HDUCLAS2"] = "OBS"
+
+        return table
+
+    def make_hdu_table(self):
+        rows = []
+        for path in self.paths:
+            rows.extend(self.get_hdu_table_rows(path))
+
+        names = list(rows[0].keys())
+        # names = ['OBS_ID', 'HDU_TYPE', 'HDU_CLASS', 'FILE_DIR', 'FILE_NAME', 'HDU_NAME']
+
+        table = Table(rows=rows, names=names)
+
+        meta = table.meta
+        meta["HDUCLASS"] = "GADF"
+        meta[
+            "HDUDOC"
+        ] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+        meta["HDUVERS"] = "0.2"
+        meta["HDUCLAS1"] = "INDEX"
+        meta["HDUCLAS2"] = "HDU"
+
+        return table
+
+    def get_hdu_table_rows(self, path):
+        events_info = self.get_events_info(path)
+        yield dict(
```

Comment:
I like to use `yield`, it's the right tool for this job IMO. I already used `yield` for all the checkers, see code further up in this file. When introducing it I was thinking if we should avoid it, for the reason you point out, to only use a subset of Python features in Gammapy and to keep the codebase easy to understand for non-Python experts.

Is your suggestion to avoid `yield` generally in Gammapy?

> Why not create a table for each observation and then later call vstack() to join the tables?

Is this really better? It will be a bit more boilerplate code to write, and on execution just do Python -> Numpy -> Python -> Numpy instead of Python -> Numpy like we do now.

Reply:
I keep using yield for now (already introduced a while ago, already used in this same file). It's internal, not part of the public API.
@adonath - if you really want to get rid of generators in Gammapy, please make the suggestion in a separate issue and label it "coding sprint" and then we discuss & decide there.